doc/src/manual/metaprogramming.md

```diff
@@ -11,6 +11,14 @@ code in Julia are represented by Julia data structures, powerful [reflection](ht
 capabilities are available to explore the internals of a program and its types just like any other
 data.
 
+!!! warning
+    Metaprogramming is a powerfull tool, but introduces complexity that makes it more difficult
+    to understand source code. It should typically be used only when other approaches scuch as 
+    [higher order faunctions](@ref man-anonymous-functions) cannot be applied.
```

Comment:
typo on `functions`

Reply:
Also on powerful

Maybe note here also that they can be surprisingly hard to get scope rules correct, and the best macros often implement most of their functionality in runtime helper functions, striving to minimize the size of the code they produce

Reply:
Otherwise, I feel that "introduces complexity" may get read as a subjective judgmental statement, which doesn't help clarify either why it is more complex (the user may still require understanding of what it calls, but cannot as easily see it) or what the alternatives are (usually closures)

Reply:
Also a typo on such

Reply:
I think I fixed all the typos.